miner/worker.go

```diff
@@ -1076,16 +1081,89 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 // into the given sealing block. The transaction selection and ordering strategy can
 // be customized with the plugin in the future.
 func (w *worker) fillTransactions(interrupt *int32, env *environment) error {
-	// Split the pending transactions into locals and remotes
-	// Fill the block with all available pending transactions.
+	// First pull Electroneum Ltd transactions out of 'pending' if flag --prioritiseElectroneum was passed by
+	// cli argument and then Split the pending transactions into locals and remotes
+	// pending is a local copy of the pool not a pointer, so it's ok to strip out Electroneum tx from localTxs and
+	// remoteTxs before processing them separately and avoid double counting.
 	pending := w.eth.TxPool().Pending(true)
+
+	electroneumTxs := make(map[common.Address]types.Transactions)
+	if w.config.PrioritiseElectroneum {
+
+		//todo: move this into worker or somewhere to save doing this every time we fill a block
+		//testing keys only
+		// private key : 8dd88ba44afa275e0dd9be1d633732242d84ea955e453b28719d17897035b4c2
+		//NB the 0x04 is to tell us that the key is uncompressed 64bytes +1 byte
+		etnSigningPubKey := "04" + "d6e7d6da18e8cfb9cf20bbb7182ac36deb9b77f2c617f3698fc12aebf22ee9b00060775349a2312813818a9c7a0a4b143fddfa36611129daf01c26bc9080d418"
+		byteStringETNKey, err := hex.DecodeString(etnSigningPubKey)
+		if err != nil {
+			fmt.Println("Error decoding hex string:", err)
+			panic("Error decoding etn signing key")
+		}
+
+		// If the first transaction for a given account has the Electroneum Ltd signature present, then all of the tx
+		// for that account do, so we can just pull that account out of pending without further thought
+		for account, txs := range pending {
+			// Extract the nonce - it's a uint64, so we need to convert it to bytes
+			nonceHex := hexutil.EncodeUint64(txs[0].Nonce())
+			nonce, _ := hex.DecodeString(nonceHex[2:]) // [2:] to trim the '0x' prefix
+			// Create signer depending on whether the transaction is protected
+			var signer types.Signer
+			if txs[0].Protected() {
+				signer = types.NewEIP155Signer(w.chainConfig.ChainID)
+			} else {
+				signer = types.HomesteadSigner{}
+			}
+
+			// Extract the sender - the function returns an address, we convert it to bytes
+			var sender []byte
+			if from, err := types.Sender(signer, txs[0]); err != nil {
+				panic(fmt.Sprintf("Failed to get transaction sender: %v", err))
```

Comment:
don't think `panic` is necessary here. if it fails in this step, let it continue the worker flow, ie. keep it in `pending` by replacing `panic` with `continue`.